plugins/palo_alto_cortex_xdr/icon_palo_alto_cortex_xdr/tasks/monitor_alerts/task.py

```diff
@@ -0,0 +1,316 @@
+
+            return response, state, has_more_pages, 200, None
+
+        except PluginException as error:
+            if isinstance(error.data, Response):
+                # if there is response data in the error then use it in the exception
+                status_code = error.data.status_code
+                cause = error.data.text
+            else:
+                status_code = 500
+                cause = error.cause
+
+            self.logger.error(
+                f"A PluginException has occurred. Status code {status_code} returned. Error: {cause}. "
+                f"Existing state: {existing_state}"
+            )
+            return [], existing_state, False, status_code, PluginException(cause=cause, data=error)
+        except Exception as error:
+            self.logger.error(
+                f"Unknown exception has occurred. No results returned. Error: {error} "
+                f"Existing state: {existing_state}"
+            )
+            return [], existing_state, False, 500, PluginException(preset=PluginException.Preset.UNKNOWN, data=error)
+
+    ###########################
+    # Make request
+    ###########################
+    def get_alerts_palo_alto(self, state: dict, start_time: int, end_time: int, alert_limit: int):
+        """ """
+
+        search_from = state.get(LAST_SEARCH_TO, 0)
+
+        search_to = search_from + alert_limit
+
+        filters = []
+
+        # If time constraints have been provided for the request, add them to the post body
+        if start_time is not None and end_time is not None:
+            filters.append({"field": self.time_sort_field, "operator": "gte", "value": start_time})
+            filters.append({"field": self.time_sort_field, "operator": "lte", "value": end_time})
+
+        post_body = self.build_post_body(search_from=search_from, search_to=search_to, filters=filters)
+
+        results, results_count, total_count = self.connection.xdr_api.get_response_alerts(post_body)
+
+        state[CURRENT_COUNT] = state.get(CURRENT_COUNT, 0) + results_count
+
+        new_alerts, last_alert_hashes, last_alert_time = self._dedupe_and_get_highest_time(results, start_time, state)
+
+        is_paginating = state.get(CURRENT_COUNT) < total_count
+
+        has_more_pages = False
+
+        if is_paginating:
+            has_more_pages = True
+            self.logger.info(f"Found total alerts={total_count}, limit={alert_limit}, is_paginating={is_paginating}")
+            self.logger.info(
+                f"Paginating alerts: Saving state with existing filters: "
+                f"search_from = {search_from} "
+                f"search_to = {search_to} "
+                f"current_count = {state.get(CURRENT_COUNT)} "
+                f"total_count = {total_count}"
+            )
+            state[LAST_SEARCH_TO] = search_to
+            state[LAST_QUERY_TIME] = start_time
+            state[LAST_SEARCH_FROM] = search_from
+            state[QUERY_START_TIME] = start_time
```

Comment:
Is this `state[QUERY_START_TIME]` necessary if it's the same as `state[LAST_QUERY_TIME]`? both are set to `start_time`? 

The only usage I can see is on line 213/214 but it seems like we don't need one of them

Reply:
Will look into it 👍🏻

Reply:
Just released a commit to address this 😄 
@ekelly-r7